### PR TITLE
add git-ignore-add function to git module

### PR DESCRIPTION
**Pull request by Eriner**
takes an arbitrary number of filenames as parameters and adds them to the .gitignore file in the project root.

### CODE REVIEW

**Comment by sorin-ionescu** [comment]
This seems useless. Just do `print abc[tab] >>! .gitignore`.

**Comment by Eriner** [comment]
@sorin-ionescu That is a much better way to handle it as opposed to the if statement. In regards to it being 'useless', this was created upon request of someone I work with. After creating it and adding it to my own zshrc, I found myself using it frequently.

It's very nice to be able to look at `git status` and see that you have files, and with one command, add many of them to your `.gitignore`. An example of this is if you are debugging a program with gdb (as well as many other utilities), they may end up creating many files, so it's nice to be able to add multiple files to the `.gitignore` at one time.

Plus, typing `git-i<tab> abc<tab>` is faster than typing `print abc<tab> >>! $(git-root)/.gitignore`

File: modules/git/functions/git-ignore-add [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#
+# Adds files passed as parameters to .gitignore in project root
+#
+# Authors:
+#   Matt Hamilton <m@tthamilton.com>
+#
+
+# make sure we have a git-root
+if ! git-root &> /dev/null; then
+  echo 'not in a git repository'
```
Comment (sorin-ionescu) [inline modules/git/functions/git-ignore-add]: `print "$0: not in a git repository" >&2`

File: modules/git/functions/git-ignore-add [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  return 1
+fi
+
+# we are in a git repository. add parameters to .gitignore
+for file in "$@"
+do
```
Comment (sorin-ionescu) [inline modules/git/functions/git-ignore-add]: Put the `do` on the previous line.

File: modules/git/functions/git-ignore-add [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  print ${file} >>! $(git-root)/.gitignore
```
Comment (sorin-ionescu) [inline modules/git/functions/git-ignore-add]: `print "$file"`

**Comment by sorin-ionescu** [comment]
You might want to make it smarter to support more [gitignore](https://git-scm.com/docs/gitignore) syntax.

**Comment by Eriner** [comment]
are you referencing supporting things like wildcard patterns? This does that. It is, afaik, impossible to have a script un-glob input, as it is parsed by the shell first. If you want to insert a wildcard, do: `git-ignore-add '*.txt'`. Making it 'smarter' in this case seems silly, as it would have to parse the current directory, and guess at what the user wants. Assuming pattern matching is what you are referencing.

**Pull request closed without merging**